### ISSUE TITLE
AMAX does not conform to netlib BLAS standard

### ISSUE THREAD

**Issue by aarongreig** [CNugteren/CLBlast#377]
When the highest value occurs more than once in an input to `AMAX` the index returned is to the last of the occurrences, but according to the netlib BLAS [documentation](http://www.netlib.org/lapack/explore-html/d0/d73/group__aux__blas_ga16c36ed9a25ca6e68931c4a00d2778e5.html#ga16c36ed9a25ca6e68931c4a00d2778e5) the expected behaviour is to return the index of the first occurrence of the highest value. I did notice that the CLBlast documentation for `AMAX` doesn't include this language, so I guess this isn't unintended behaviour in the design of the `Xamax` kernel. In case you think the `Xamax` kernel should be returning the first index I've included my clinfo output in clinfo.txt so you can see what drivers I'm running. I see this behaviour with both pocl and intel.
[clinfo.txt](https://github.com/CNugteren/CLBlast/files/4276478/clinfo.txt)

**Comment by CNugteren** [comment]
Hmm, you might be right. Could you check if the behaviour is indeed as expected if you change the `>=` into an `>` in the following places:
* https://github.com/CNugteren/CLBlast/blob/master/src/kernels/level1/xamax.opencl#L67
* https://github.com/CNugteren/CLBlast/blob/master/src/kernels/level1/xamax.opencl#L80
* https://github.com/CNugteren/CLBlast/blob/master/src/kernels/level1/xamax.opencl#L108
* https://github.com/CNugteren/CLBlast/blob/master/src/kernels/level1/xamax.opencl#L121
I'm a bit afraid this is not a proper fix, since AMIN might then suffer from the opposite. But in any case the documentation should also be an updated, and it should be investigated why the tests don't fail: perhaps the tested data is such that all values are unique.

**Comment by aarongreig** [comment]
>Could you check if the behaviour is indeed as expected if you change the >= into an > in the following places:

Yeah that seems to fix it.

**Comment by aarongreig** [comment]
Also, it looks like AMIN *is* similarly affected and the s/>=/>/ fix also fixes that.

**Comment by CNugteren** [comment]
I've made a PR here: https://github.com/CNugteren/CLBlast/pull/378. However:
* I couldn't easily make the test suite catch this, because there is still no proper support in the code for integer-based results, all the tests assume floating point values. So instead I wrote a small sample to manually test for now.
* The small sample shows that the proposed solution is not solving the issue, just making it more likely to return the index of one of the first results. That's a bit due to how the reduction is performed, taking into account memory access patterns and such.

So for now I propose what's in the PR, which has updated API documentation saying that it can choose any of the maximums based on the kernel's tuning parameters.
